cmd/dcrdata/internal/middleware/apimiddleware.go

```diff
@@ -1101,3 +1102,24 @@ func RetrieveStickWidthCtx(r *http.Request) string {
 	}
 	return bin
 }
+
+// APIVersionCtx adds supported API version to a request context.
+func APIVersionCtx(version string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			r = r.WithContext(context.WithValue(r.Context(), ctxAPIVersion, version))
```

Comment:
Why would this not be an integer?

Reply:
I think it will read better as `v1, v2` when used by request handlers, but it can be an integer.

Reply:
hander logic:

```go
switch verCtx {
case 0:
...
case 1:
...
default:
...
}
```

better than string matching.